utils/src/main/scala/za/co/absa/enceladus/utils/general/Section.scala

```diff
@@ -0,0 +1,288 @@
+    * @param that the other Section
+    * @return     true if the greater Section starts before the smaller one ends (see comparison) or right after it
+    *             false otherwise
+    */
+  def touches(that: Section): Boolean = {
+    distance(that).exists(_ <= 0)
+  }
+}
+
+object Section {
+  def fromIndexes(start: Int, end: Int): Section = {
+    Section(start, end - start + 1)
+  }
+
+  def ofSameChars(fromString: String, fromIndex: Int): Section = {
+    val realFromIndex = if (fromIndex >= 0) {
+      fromIndex
+    } else {
+      fromString.length + fromIndex
+    }
+    if ((realFromIndex >= fromString.length) || (realFromIndex < 0)) {
+      Section(fromIndex, 0)
+    } else {
+      val char = fromString(realFromIndex)
+      var res = realFromIndex
+      while ((res < fromString.length) && (fromString(res) == char)) {
+        res += 1
+      }
+      Section(fromIndex, res - realFromIndex)
+    }
+  }
+
+  /**
+    * Removes sections of the string in a way, that string is considered intact until all removals are executed. In
+    * other words the indexes are not shifted.
+    * @param fromString the string to operate upon
+    * @param sections   sections to apply
+    * @return           the string as a result if all the sections would be removed "at once"
+    */
+  def removeMultiple(fromString: String, sections: Seq[Section]): String = {
+    if (sections.isEmpty) {
+      fromString
+    } else {
+      val charsPresent = Array.fill(fromString.length)(true)
+      sections.foreach{section =>
+        val (realStart, after) = section.toSubstringParameters(fromString)
+        for (i <- realStart until after) {
+          charsPresent(i) = false
+        }
+      }
+      val paring: Seq[(Char, Boolean)] = fromString.toSeq.zip(charsPresent)
+
+      paring.collect{case (c, true) => c}.mkString
+    }
+  }
+
+  /**
+    * Merges all touching (that includes overlaps too) sections and sort them
+    * @param sections the sections to merge
+    * @return         an ordered from greater to smaller sequence of distinct sections (their distance is at least 1 or undefined)
+    */
+  def mergeSections(sections: Seq[Section]): Seq[Section] = {
+    def fuse(into: Section, what: Section): Section = {
+      if (into.start + into.length >= what.start + what.length) {
+        //as the sequence where the sections are coming from is sorter, this condition is enough to check that `what` is within `into`
+        into
+      } else {
+        //actual fusion
+        //the length expression is simplified: into.length + what.length - ((into.start + into.length) - what.start)
+        Section(into.start, what.length - into.start + what.start)
+      }
+    }
+
+    def doMerge(input: Seq[Section]): Seq[Section] = {
+      if (input.isEmpty) {
+        input
+      } else {
+        val sorted = input.sorted
+        sorted.tail.foldLeft(List(sorted.head)) { (resultAcc, item) =>
+          if (item touches resultAcc.head) {
```

Comment:
please change from infix notation to prefix notation
```suggestion
          if (item.touches(resultAcc.head)) {
```

Reply:
I have to admit I really like the _inflix_ here. Makes it easy to read, no issues with precedence. You mind it much? 😉

Reply:
I don't _mind it_, I would just keep everything the "same".